android/source/VideoLocker/src/org/edx/mobile/base/BaseFragmentActivity.java

```diff
@@ -66,7 +66,7 @@
     protected IStorage storage;
     protected ISegment segIO;
     protected boolean runOnTick = true;
-    protected final Logger logger = new Logger(getClass().getName());
+    protected transient final Logger logger = new Logger(getClass().getName());
```

Comment:
In case of Android Activities, when screen rotates, say from portrait to landscape, all the fields in the class are serialized and de-serialized during restore. This requires all the fields in the class to be <code>Serializable</code>. Activities do this, though not explicitly marked as <code>Serializable</code>. 

If we want to skip any field from being serialized, we have to mark it as transient. We have only a few screens which host player and hence support orientation change. This is where we got an exception.

I changed <code>Logger</code> class to implement <code>Serializable</code>, so we don't really need to use transient keyword here. I will remove transient keyword from the <code>Logger</code> instance creation.